File: ApiScript.py
from IronWASP import *

def start_log_range(req, res):
	GlobalStore.Put("log_start_py_api", Config.LastProxyLogId + 1)
	res.BodyString = "OK"
	
ApiCallHandler.AddHandler("/py/log_range_start", start_log_range)
	
def end_log_range(req, res):
	GlobalStore.Put("log_end_py_api", Config.LastProxyLogId)
	res.BodyString = "OK"

ApiCallHandler.AddHandler("/py/log_range_end", end_log_range)
	
def scan_log_range(req, res):
	start_id = GlobalStore.Get("log_start_py_api")
	end_id = GlobalStore.Get("log_end_py_api")
	for i in range(start_id, end_id + 1):
		r = Request.FromProxyLog(i)
		s = Scanner(r)
		if r.Query.Count > 0 or r.Body.Count > 0:
			s.InjectQuery()
			s.InjectBody()
			s.CheckAll()
			s.LaunchScan()
	res.BodyString = "OK"

ApiCallHandler.AddHandler("/py/scan_log_range", scan_log_range)


def connection_check(req, res):
	res.BodyString = "OK"

ApiCallHandler.AddHandler("connection_check", connection_check)

def passpharse_check(req, res):
	res.BodyString = "OK"

ApiCallHandler.AddHandler("passpharse_check", connection_check)

File: ApiScript.rb
include IronWASP

def start_log_range(req, res)
	GlobalStore.put("log_start_rb_api", Config.last_proxy_log_id + 1)
	res.body_string = "OK"
end
	
start_log_range_l = lambda{|req, res| start_log_range(req, res)}
ApiCallHandler.add_handler("/rb/log_range_start", start_log_range_l)
	
def end_log_range(req, res)
	GlobalStore.put("log_end_rb_api", Config.last_proxy_log_id)
	res.body_string = "OK"
end

end_log_range_l = lambda{|req, res| end_log_range(req, res)}
ApiCallHandler.add_handler("/rb/log_range_end", end_log_range_l)
	
def scan_log_range(req, res)
	start_id = GlobalStore.get("log_start_rb_api")
	end_id = GlobalStore.get("log_end_rb_api")
	(start_id..end_id).each do |i|
		r = Request.from_proxy_log(i)
		s = Scanner.new(r)
		if r.query.count > 0 or r.body.count > 0
			s.inject_query
			s.inject_body
			s.check_all
			s.launch_scan
		end
	end
	res.body_string = "OK"
end

scan_log_range_l = lambda{|req, res| scan_log_range(req, res)}
ApiCallHandler.add_handler("/rb/scan_log_range", scan_log_range_l)
